src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -458,7 +436,7 @@ public override BoundStatement InstrumentUsingTargetCapture(BoundUsingStatement
 
         private BoundStatement AddDynamicAnalysis(BoundStatement original, BoundStatement rewritten)
         {
-            if (!original.WasCompilerGenerated)
+            if (!original.WasCompilerGenerated && !_methodBodyFactory.IsCodeCoverageDisabled)
```

Comment:
> && !_methodBodyFactory.IsCodeCoverageDisabled [](start = 46, length = 46)

It doesn't look like this is going to intercept all scenarios where we add instrumentation. For example, there is a code path in InstrumentReturnStatement that doesn't come through here. InstrumentSwitchWhenClauseConditionalGotoBody also does its own instrumentation. Etc. I think we should look for a good place where we can intercept all attempts to inject instrumentation in the way where we wouldn't have to keep checking per construct, in case new specialized instrumentation scenarios would come up in the future.